generator.rb

```diff
@@ -0,0 +1,111 @@
+def is_vowel(letter)
+  case letter
+  when "a", "e", "i", "o", "u"
+    return "true"
+  else
+    return "false"
+  end
+end
+
+def rev_str(strg)
+  aa = strg.split("")
+  rev = Array.new
+  aa.length.times do
+    rev << aa.pop
+  end
+  rev = rev.join
+  return rev
+end
+
+def check_length(word)
+  if word.length < 2
+    until word.length > 1
+      puts "Your word must be 2 or more characters."
+      word = gets.chomp
+    end
+    return word
+  else
+    return word
+  end
+end
+
+def vowel_index(word)
+  count = 0
+  word.each_char { |letter|
+    if is_vowel(letter) == "false"
+      count += 1
+    else
+      break
+    end
+  }
+  return count
+end
+
+def check_contain_vowel(word)
+  count = vowel_index(word)
+  if count == word.length
+    until count != word.length
```

Comment:
Again, you don't need both the `if` and the `until`, since they're checking the same thing. If `count != word.length`, the `until` loop won't run at all.